otp/imx6ull-otp/imx6ull-otp.c

```diff
@@ -567,36 +588,34 @@ int main(int argc, char **argv)
 			common.get_uid = 1;
 			break;
 		case 'm':
-			if (argc < 5) {
-				common.display_usage = 1;
-				break;
+			if (argc - (optind - 1) < 3) {
+				usage(argv[0]);
+				return 1;
 			}
+			/* WARN: using multiple positional arguments with getopt options is implementation-defined */
 			mac[0] = argv[optind - 1];
 			if (argv[optind] == NULL || argv[optind][0] == '-') {
-				common.display_usage = 1;
-				break;
+				usage(argv[0]);
+				return 1;
```

Comment:
We rather not care about mmaped memory? If we do, maybe setting some flag `fail = 1; break;` and exits handled in possibly one place would be better.

Reply:
moved `mmap` to after input parameter validation, we have one exit path there so it's easierl.